packages/dd-trace/src/debugger/devtools_client/index.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const { randomUUID } = require('crypto')
+const { breakpoints } = require('./state')
+const session = require('./session')
+const send = require('./send')
+const { ackEmitting } = require('./status')
+const { parentThreadId } = require('./config')
+const log = require('../../log')
+const { version } = require('../../../../../package.json')
+
+require('./remote_config')
+
+// There doesn't seem to be an official standard for the content of these fields, so we're just populating them with
+// something that should be useful to a Node.js developer.
+const threadId = parentThreadId === 0 ? `pid:${process.pid}` : `pid:${process.pid};tid:${parentThreadId}`
+const threadName = parentThreadId === 0 ? 'MainThread' : 'WorkerThread'
+
+session.on('Debugger.paused', async ({ params }) => {
+  const start = process.hrtime.bigint()
+  const timestamp = Date.now()
+  const probes = params.hitBreakpoints.map((id) => breakpoints.get(id))
+  await session.post('Debugger.resume')
+  const diff = process.hrtime.bigint() - start // TODO: Should this be recored as telemetry?
+
+  log.debug(`Finished processing breakpoints - main thread paused for: ${Number(diff) / 1000000} ms`)
+
+  const logger = {
+    // We can safely use `location.file` from the first probe in the array, since all probes hit by `hitBreakpoints`
+    // must exist in the same file since the debugger can only pause the main thread in one location.
+    name: probes[0].location.file, // name of the class/type/file emitting the snapshot
+    method: params.callFrames[0].functionName, // name of the method/function emitting the snapshot
+    version,
+    thread_id: threadId,
+    thread_name: threadName
+  }
+
+  // TODO: Send multiple probes in one HTTP request as an array
+  for (const probe of probes) {
+    const snapshot = {
+      id: randomUUID(),
+      timestamp,
+      probe: {
+        id: probe.id,
+        version: probe.version,
+        location: probe.location
+      },
+      language: 'javascript'
+    }
+
+    // TODO: Process template
+    send(probe.template, logger, snapshot, (err) => {
```

Comment:
how big can be the `probes` array? Just to keep in mind that there is a max of request that the tracer could do:
https://github.com/DataDog/dd-trace-js/blob/bd2f8b2b0c0cd0d35a600912b99aa1c3f7e74d6d/packages/dd-trace/src/exporters/common/request.js#L18
if `probes.length` could be bigger than 8, max of first 8 will be sent, the rest will be discarded.

Reply:
99% of the time there will just be a single probe. I don't think there will more than 2 or 3. If I understand the Inspector API correctly, it's only if you have more than a single breakpoint in the same location (file+line+ column).

In any case there's a follow up task after this PR to send them all as one request in an array,